File: android/app/src/main/java/com/fabricio/challenge/model/Category.java
package com.fabricio.challenge.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

/**
 * Model of the Category information fetch from the REST API
 * @author Fabricio Godoi
 */
public class Category {

    @Expose
    @SerializedName("id")
    private Integer id;

    @Expose
    @SerializedName("descricao")
    private String description;

    @Expose
    @SerializedName("urlImagem")
    private String urlImage;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getUrlImage() {
        return urlImage;
    }

    public void setUrlImage(String urlImage) {
        this.urlImage = urlImage;
    }
}

File: android/Lodjinha/app/src/main/java/com/example/lodjinha/data/classes/Category.kt
package com.example.alodjinha.data.classes

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

    class Category {

    @SerializedName("id")
    @Expose
    private var id: Int? = null

    @SerializedName("descricao")
    @Expose
    private var description: String? = null


    @SerializedName("urlImagem")
    @Expose
    private var urlImage: String? = null

    fun setId(Id: Int){ id = Id }

    fun getId() = id

    fun setUrlImage(UrlImage: String){ urlImage = UrlImage}

    fun getUrlImage() = urlImage

    fun setDescription(Description: String){ description = Description }

    fun getDescription() = description
}